launcher/java/JavaUtils.cpp

```diff
@@ -554,3 +558,36 @@ QStringList getPrismJavaBundle()
 
     return javas;
 }
+
+QStringList getPrismExtraJavaPaths()
+{
+    QList<QString> javas;
+
+    QString executable = "java";
+#if defined(Q_OS_WIN32)
+    executable += "w.exe";
+#endif
+
+    auto scanDir = [&](QString prefix) {
+        javas.append(FS::PathCombine(prefix, "jre", "bin", executable));
+        javas.append(FS::PathCombine(prefix, "bin", executable));
+        javas.append(FS::PathCombine(prefix, executable));
+    };
```

Comment:
That is the fun in all this: the guessing makes this worth it.
Otherwise, users can browse for the Java executable without the add path thing.
This PR is so we do not get any more of the "add this java path" PR.
But let me know if you really want to drop the `scanDir` function in favor of `javas.append(FS::PathCombine(prefix, "bin", executable));`

Reply:
I think dropping the scanDir function would make a lot more sense. Having two prefix paths be able to resolve with different suffixes is a bit odd; this should work similar to [how we handle `PRISMLAUNHCER_JAVA_PATHS`](https://github.com/PrismLauncher/PrismLauncher/blob/44b4262f3ce037140cb6415cea27ed44bd03164d/launcher/java/JavaUtils.cpp#L154), which is basically this as an env var